### ISSUE TITLE
[Bug]: V1.py(Line 234): osp.dirname(self.cache_path) cause FileNotFoundError: [WinError 3] on Windows with poetry

### ISSUE THREAD

**Issue by swiftord** [acheong08/ChatGPT#935]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits

### What happened?

V1.py(Line 234): osp.dirname(self.cache_path) cause **FileNotFoundError: [WinError 3]** on Windows with poetry

```bat
Traceback (most recent call last):
  File "D:\chatgpt\test.py", line 11, in <module>
    chatbot = AsyncChatbot(conversation_id=conversation_id, config={
  File "D:\chatgpt\.venv\lib\site-packages\revChatGPT\V1.py", line 558, in __init__
    super().__init__(
  File "D:\chatgpt\.venv\lib\site-packages\revChatGPT\V1.py", line 45, in wrapper
    out = func(*args, **kwargs)
  File "D:\chatgpt\.venv\lib\site-packages\revChatGPT\V1.py", line 147, in __init__
    self.__check_credentials()
  File "D:\chatgpt\.venv\lib\site-packages\revChatGPT\V1.py", line 45, in wrapper
    out = func(*args, **kwargs)
  File "D:\chatgpt\.venv\lib\site-packages\revChatGPT\V1.py", line 152, in __check_credentials
    self.__refresh_headers(self.config["access_token"])
  File "D:\chatgpt\.venv\lib\site-packages\revChatGPT\V1.py", line 45, in wrapper
    out = func(*args, **kwargs)
  File "D:\chatgpt\.venv\lib\site-packages\revChatGPT\V1.py", line 183, in __refresh_headers
    self.__cache_access_token(email, access_token)
  File "D:\chatgpt\.venv\lib\site-packages\revChatGPT\V1.py", line 45, in wrapper
    out = func(*args, **kwargs)
  File "D:\chatgpt\.venv\lib\site-packages\revChatGPT\V1.py", line 230, in __cache_access_token
    self.__write_cache(cache)
  File "D:\chatgpt\.venv\lib\site-packages\revChatGPT\V1.py", line 45, in wrapper
    out = func(*args, **kwargs)
  File "D:\chatgpt\.venv\lib\site-packages\revChatGPT\V1.py", line 234, in __write_cache
    os.makedirs(osp.dirname(self.cache_path), exist_ok=True)
  File "D:\Python3.10.9\lib\os.py", line 225, in makedirs
    mkdir(name, mode)
FileNotFoundError: [WinError 3] System can not find the route。: ''
```

### Steps to reproduce the problem

1. Use Windows 10 and poetry to install revchatgpt v2.3.14
2. from revChatGPT.V1 import AsyncChatbot
3. try to call "ask" function and with a conversation_id and a null parent_id

### What should have happened?

No error when revchatgpt try to write cache.

### Version where the problem happens

2.3.12~2.3.14

### What Python version are you running this with?

3.10.9

### What is your operating system ?

Windows

### Command Line Arguments

```Shell
No
```

### Console logs

```Shell
Traceback (most recent call last):
  File "D:\chatgpt\test.py", line 11, in <module>
    chatbot = AsyncChatbot(conversation_id=conversation_id, config={
  File "D:\chatgpt\.venv\lib\site-packages\revChatGPT\V1.py", line 558, in __init__
    super().__init__(
  File "D:\chatgpt\.venv\lib\site-packages\revChatGPT\V1.py", line 45, in wrapper
    out = func(*args, **kwargs)
  File "D:\chatgpt\.venv\lib\site-packages\revChatGPT\V1.py", line 147, in __init__
    self.__check_credentials()
  File "D:\chatgpt\.venv\lib\site-packages\revChatGPT\V1.py", line 45, in wrapper
    out = func(*args, **kwargs)
  File "D:\chatgpt\.venv\lib\site-packages\revChatGPT\V1.py", line 152, in __check_credentials
    self.__refresh_headers(self.config["access_token"])
  File "D:\chatgpt\.venv\lib\site-packages\revChatGPT\V1.py", line 45, in wrapper
    out = func(*args, **kwargs)
  File "D:\chatgpt\.venv\lib\site-packages\revChatGPT\V1.py", line 183, in __refresh_headers
    self.__cache_access_token(email, access_token)
  File "D:\chatgpt\.venv\lib\site-packages\revChatGPT\V1.py", line 45, in wrapper
    out = func(*args, **kwargs)
  File "D:\chatgpt\.venv\lib\site-packages\revChatGPT\V1.py", line 230, in __cache_access_token
    self.__write_cache(cache)
  File "D:\chatgpt\.venv\lib\site-packages\revChatGPT\V1.py", line 45, in wrapper
    out = func(*args, **kwargs)
  File "D:\chatgpt\.venv\lib\site-packages\revChatGPT\V1.py", line 234, in __write_cache
    os.makedirs(osp.dirname(self.cache_path), exist_ok=True)
  File "D:\Python3.10.9\lib\os.py", line 225, in makedirs
    mkdir(name, mode)
FileNotFoundError: [WinError 3] System can not find the route。: ''
```

**Comment by Full4me** [comment]
How did you solve this problem?

**Comment by PowermateEx** [comment]
> @Full4me revChatGPT Optional configuration said "If you are using Windows, you will need to create an environment variable named HOME and set it to your home profile for the script to be able to locate the config.json file."
> 
> I haven't tried it yet, but maybe this isn't a bug, so I just closed this issue.

Where is the config.json supposed to be? I've set the HOME environment, but it seems to be still wrong and i dont know where this config.json is. Would you care to explain to someone who doesnt know all the ins and outs of that? Thanks

**Comment by swiftord** [comment]
@PowermateEx It seems that this is indeed a bug, and it has been fixed in version 2.3.15 and 2.3.16, you can try to update your local package version

**Comment by PowermateEx** [comment]
Okay never mind i figured it out myself by trial and error, lol.
i've set it to "C:\Users\UserName\AppData\Local" and it works now.
